### PR TITLE
feat(cli): db list in reverse

### PR DESCRIPTION
**Pull request by rakita**
Added `--reverse` option to `db list` cli command, this will show last entries from the table.

Additionaly renames `--start` to `--skip` to better reflects what is going on.

And fixed small bug where we could potentially underflow `skip` (previously `start`) integer.

### CODE REVIEW

File: bin/reth/src/db/mod.rs [hunk 1]
```diff
@@ -72,9 +72,12 @@ pub enum Subcommands {
 pub struct ListArgs {
     /// The table name
     table: String, // TODO: Convert to enum
-    /// Where to start iterating
+    /// Skip first N entries
     #[arg(long, short, default_value = "0")]
-    start: usize,
+    skip: usize,
```
Comment (mattsse) [inline bin/reth/src/db/mod.rs:77]: ah right, skip is the same is start in normal order

**Review by mattsse — APPROVED** [review]
all of this makes sense to me

File: bin/reth/src/db/mod.rs (hunk 1 above)
Comment (rakita) [reply bin/reth/src/db/mod.rs:77]: It was just a simple rename

**Comment by rakita** [comment]
It seems there is some bug with duptables, so let me check that

**Comment by rakita** [comment]
> It seems there is some bug with duptables, so let me check that

found it, it now works with dupsort tables. Previously we used `BTreeMap` to store entries that would just overwrite keys for duplicated values. Replacment BTreeMap to Vec solved the problem

**Comment by rakita** [comment]
Tested, should be fine to merge after ci finishes

**Pull request merged**